ReactAndroid/src/main/java/com/facebook/react/jstasks/HeadlessJsTaskContext.java

```diff
@@ -105,7 +118,44 @@ public synchronized int startTask(final HeadlessJsTaskConfig taskConfig) {
     for (HeadlessJsTaskEventListener listener : mHeadlessJsTaskEventListeners) {
       listener.onHeadlessJsTaskStart(taskId);
     }
-    return taskId;
+  }
+
+  /**
+   * Retry a running JS task with a delay. Invokes
+   * {@link HeadlessJsTaskContext#startTask(HeadlessJsTaskConfig, int)} as long as the process does
+   * not get killed.
+   *
+   * @return true if a retry attempt has been posted.
+   */
+  public synchronized boolean retryTask(final int taskId) {
+    final HeadlessJsTaskConfig sourceTaskConfig = mActiveTaskConfigs.get(taskId);
+    Assertions.assertCondition(
+      sourceTaskConfig != null,
+      "Tried to retrieve non-existent task config with id " + taskId + ".");
+
+    final HeadlessJsTaskRetryPolicy retryPolicy = sourceTaskConfig.getRetryPolicy();
+    if (!retryPolicy.canRetry()) {
+      return false;
+    }
+
+    removeTimeout(taskId);
+    final HeadlessJsTaskConfig taskConfig = new HeadlessJsTaskConfig(
+            sourceTaskConfig.getTaskKey(),
+            sourceTaskConfig.getData(),
+            sourceTaskConfig.getTimeout(),
+            sourceTaskConfig.isAllowedInForeground(),
+            retryPolicy.update()
+    );
+
+    final Runnable retryAttempt = new Runnable() {
+      @Override
+      public void run() {
+        startTask(taskConfig, taskId);
```

Comment:
Do we reuse the taskId just so JS can continue to use the same id? Or to avoid adding entries to the active tasks map? (I think it makes sense to use the same taskId, just curious if there's another reason why it's needed)